es6-async-await/exceptions-best.js

```diff
@@ -0,0 +1,35 @@
+import fetch from './fetch.js';
+
+const startTime = Date.now();
+const elapsed = () => `${Math.round((Date.now() - startTime) / 1000)}s -`;
+
+async function throwOnce() {
+  const msg = await fetch('foo', false);
+  console.log(elapsed(), 'throwOnce:', msg);
+}
+
+async function throwSeveral() {
+  const msg1 = await fetch('foo1', false);
+  console.log(elapsed(), 'throwSeveral1:', msg1);
+  const msg2 = await fetch('foo2', false);
+  console.log(elapsed(), 'throwSeveral2:', msg2);
+  const msg3 = await fetch('foo3', false);
+  console.log(elapsed(), 'throwSeveral3:', msg3);
+}
+
+async function throwChained() {
+  const msg1 = await fetch('foo-chain', false);
+  console.log(elapsed(), 'throwChained1:', msg1);
+  const msg2 = await fetch(msg1, false);
+  console.log(elapsed(), 'throwChained2:', msg2);
+  const msg3 = await fetch(msg2, false);
+  console.log(elapsed(), 'throwChained3:', msg3);
+}
+
+try {
+  throwOnce()
+    .then(() => throwSeveral())
+    .then(() => throwChained());
+} catch (error) {
+  console.log('Error:', error.message);
+}
```

Comment:
Since there is no `await`, the `throwXxx` calls will never throw an exception. So this error handling doesn't work. You should be able to just use `await` on each of these instead of using Promise.then. If for some reason you can't use `await` here (such as if you have an older version of Node), then you can remove the try/catch and use `Promise.then().then().catch()`.